plugins/inputs/mavlink/mavlink.go

```diff
@@ -0,0 +1,196 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mavlink
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/bluenviron/gomavlib/v3"
+	"github.com/bluenviron/gomavlib/v3/pkg/dialects/ardupilotmega"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Mavlink struct {
+	URL                    string   `toml:"url"`
+	SystemID               uint8    `toml:"system_id"`
+	FilterPattern          []string `toml:"filter"`
+	StreamRequestEnable    bool     `toml:"stream_request_enable"`
+	StreamRequestFrequency int      `toml:"stream_request_frequency"`
```

Comment:
Shouldn't this be an unsigned type? If not, please at least check for negative values in `Init()`!